libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/headline/headlineBucketConversion.ts

```diff
@@ -0,0 +1,59 @@
+// (C) 2023 GoodData Corporation
+import {
+    IInsightToPropConversion,
+    PropMeta,
+    sdkModelPropMetas,
+} from "../../../utils/embeddingCodeGenerator/index.js";
+import { bucketMeasure, bucketMeasures, IBucket, IMeasure, insightBucket } from "@gooddata/sdk-model";
+import { IEmbeddingCodeContext } from "../../../interfaces/VisualizationDescriptor.js";
+
+export function singleSecondaryMeasureBucketConversion<TProps extends object, TPropKey extends keyof TProps>(
+    propName: TPropKey,
+    bucketName: string,
+): IInsightToPropConversion<TProps, TPropKey, IMeasure> {
+    return bucketConversion(
+        propName,
+        sdkModelPropMetas.Measure.Single,
+        bucketName,
+        (ctx) => {
+            return !ctx?.settings?.enableNewHeadline;
+        },
+        bucketMeasure,
+    );
+}
+
+export function multipleSecondaryMeasuresBucketConversion<
+    TProps extends object,
+    TPropKey extends keyof TProps,
+>(propName: TPropKey, bucketName: string): IInsightToPropConversion<TProps, TPropKey, IMeasure[]> {
+    return bucketConversion(
+        propName,
+        sdkModelPropMetas.Measure.Multiple,
+        bucketName,
+        (ctx) => {
+            return ctx?.settings?.enableNewHeadline;
+        },
+        bucketMeasures,
+    );
+}
+
+export function bucketConversion<
+    TProps extends object,
+    TPropKey extends keyof TProps,
+    TReturnType = TProps[TPropKey],
+>(
+    propName: TPropKey,
+    propType: PropMeta,
+    bucketName: string,
+    settingCondition: (ctx: IEmbeddingCodeContext) => boolean,
```

Comment:
I would name it `isSettingEnabled` or `isSettingSupported` based on the context of what is supposed to do. `settingCondition` seems bit cryptic to me. The name of the function variable should state what the condition is supposed to be for.